### PR TITLE
[92] progress without error

### PR DESCRIPTION
**Pull request by amyoulton**
This PR adds in the ability for a request to have a postProcessing method. To implement it, I have done the following:

1. Added `postProcess` method to Netable request, that by default returns the `FinalResource`. This will now be the final method run through every request.
2. If the user adds a post process method to the request, something can be run before the `FinalResource` is returned.
3. I have added in a simple example to the `Example` project, where we have a Data Manager that stores a user. The `GetUserRequest` implements the `postProcess` method, where we send the user to the shared user object in the DataManager before returning the finalResource.
4. In Homeview, I added a print method to showcase this working.

Note: I also removed the target here for the old example. I _think_ that was also done in the sendable ticket which hasn't been merged, but if not it's done here now! Whichever ticket is merged first, I will rebase the other, so it shouldn't matter.

### CODE REVIEW

**Review by brendanlensink — CHANGES_REQUESTED** [review]
A couple suggestions/comments to think about

Also we should probably bump the library version since we're adding new functionality ✨

File: Netable/Netable/Request.swift [hunk 1]
```diff
@@ -59,6 +59,9 @@ public protocol Request: Sendable {
 
     /// Optional: The method to convert your RawResource returned by the server to FinalResource.
     func finalize(raw: RawResource) async throws -> FinalResource
+
+    /// Optional: The method to perform a process after getting the result but before the request returns it.
+    func postProcess(result: FinalResource) async throws -> FinalResource
```
[the remark this answers is not in the document]
Comment (brendanlensink) [reply Netable/Netable/Request.swift]: @nbrooke it looks like you suggested the return value in the initial suggestion, is there a reason you thought we should keep it around?

[the remark this answers is not in the document]
Comment (nbrooke) [reply Netable/Netable/Request.swift]: Having mutable structs (or actors) be Sendable is totally fine (as long as all the members are Sendable). It's only classes where you might need to make things immutable to get sendability. So the postProcess hook CAN modify some types, and it seems valuable to allow it as an option.

[the remark this answers is not in the document]
Comment (brendanlensink) [reply Netable/Netable/Request.swift]: Fair enough, sorry for the confusion @amyoulton. Lets back out of that change 😬

**Review by brendanlensink — APPROVED** [review]
Looks good, before merging don't forget to add a blurb to the README explaining intended use 🚀

**Comment by amyoulton** [comment]
@brendanlensink this is merged! You said you were going to do the tag and release stuff so just nudging about that! 👀

**Pull request merged**